apis/tempo/v1alpha1/tempostack_webhook.go

```diff
@@ -146,6 +146,12 @@ func (d *Defaulter) Default(ctx context.Context, obj runtime.Object) error {
 		r.Spec.Template.QueryFrontend.JaegerQuery.Ingress.Route.Termination = TLSRouteTerminationTypeEdge
 	}
 
+	// if tenant mode is Openshift, ingress type should be route by default.
+	if r.Spec.Tenants != nil && r.Spec.Tenants.Mode == OpenShift && r.Spec.Template.Gateway.Ingress.Type == IngressTypeRoute {
+		r.Spec.Template.Gateway.Ingress.Type = IngressTypeRoute
```

Comment:
afaics `r.Spec.Template.Gateway.Ingress.Type == IngressTypeRoute` should be removed from the condition